src/katgpucbf/xbgpu/engine.py

```diff
@@ -148,8 +154,15 @@ class Pipeline:
     :class:`.QueueItem` provides mechanisms to wait for the in-progress work to
     complete.
 
+    .. todo::
+        Update the `output` parameter to be a list of `Output`s once multiple
+        beams are supported. This will require some updates to the plumbing
+        that uses `output` to e.g. capture-{start, stop}.
+
     Parameters
     ----------
+    output
+        Output config for data product (BOutput or XOutput).
```

Comment:
I'd recommend against adding this - you're likely to paint yourself into a corner where it becomes messy to remove it again to support multiple beams.

Instead, I'd suggest making it plural from the start (outputs: Sequence[Output]). In XPipeline you can then have `output` be an alias to `outputs[0]` (e.g. as a `@property`).